pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UseUnderscoresInNumericLiterals.xml

```diff
@@ -0,0 +1,318 @@
+        </description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = -0b100111;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>
+            ok, Numeric Literal in negative octal
+        </description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = -047474747;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>
+            ok, Numeric Literal in negative hexadecimal
+        </description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = -0xabcdef;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>
+            ok, Numeric Literal with exponent
+        </description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = 39e1234;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>
+            bad, Negative exponent without '_'
+        </description>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = 36e-11234;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>
+            ok, Numeric Literal with positive exponent
+        </description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = 100e+10;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>
+            ok, Lengthy numeric literal with variable name as serialVersionUID
+        </description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    static final long serialVersionUID = -2518308549741147689L;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>
+            bad, testing acceptableDecimalLength
+        </description>
+        <rule-property name="acceptableDecimalLength">3</rule-property>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    private callBar() {
+        bar(1000);
+    }
+}
+     ]]></code>
+    </test-code>
+    <test-code>
+        <description>
+            ok, testing acceptableDecimalLength
+        </description>
+        <rule-property name="acceptableDecimalLength">5</rule-property>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int value = 10000;
+}
+     ]]></code>
+    </test-code>
+    <test-code>
+        <description>
+            bad, testing acceptableDecimalLength
+        </description>
+        <rule-property name="acceptableDecimalLength">10</rule-property>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int value = -32_3639_6_9;
```

Comment:
This test should test a number that is under the acceptableDecimalLength to check that misplaced underscores are flagged. This one is 11 chars long ^^

Reply:
are you thinking of a test case similar to this one - https://github.com/pmd/pmd/pull/1384/files/9b6b1a4d15b7013b302467edab093f941f5d8fc4#diff-23912dc999320f5a64c4d06d86bdfdd2R96 ?
This one does not have acceptableDecimalLength defined, so it gets defaulted to 4.

Reply:
In fact the test case you linked tests exactly the same thing as this one, since the number is longer than the acceptable length and the underscores are misplaced in both cases. I'd like a test checking that misplaced underscores are flagged even if the length is acceptable. eg `acceptableDecimalLength=10` and test on `3639_639` (length 8)

Reply:
my bad! I got confused. Modified the length now.